data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/router/RouterFactory.java

```diff
@@ -22,6 +22,6 @@ public class RouterFactory {
 
     public Router createRouter(final Set<ConditionalRoute> routes) {
         final RouteEventEvaluator routeEventEvaluator = new RouteEventEvaluator(expressionEvaluator, routes);
-        return new Router(routeEventEvaluator, dataFlowComponentRouter);
+        return new Router(routeEventEvaluator, dataFlowComponentRouter, event -> event.getEventHandle().release(true));
```

Comment:
Let's add a test case for this in `RouterFactoryTest`.

You can do this with some mocking:

1. Mock the constructor for `Router`.
2. Call `createRouter`.
3. Extract the `Consumer` provided as an argument to the constructor.
4. Call the real consumer with a mocked `Event` and verify the expected calls.